syntax = "proto3";

package gemm_simple;


message MatMulInitRequest {
	int32 m = 1;
	int32 k = 2;
	int32 n = 3;
}

message MatMulInitResponse {
	int32 operationId = 1;
}

message MatricesInput {
	int32 operationId = 1;
	int32 m = 2;
	int32 k = 3;
	int32 n = 4;
	repeated float data = 5;
}

message MatrixOutput {
	int32 operationId = 1;
	int32 m = 2;
	int32 n = 3;
	repeated float data = 4;
}

service MatMul{
	rpc RequestMatMul(MatMulInitRequest) returns (MatMulInitResponse);
	rpc MatrixMultiply(MatricesInput) returns (MatrixOutput);
}